handler/src/main/java/io/netty/handler/ssl/SslHandler.java

```diff
@@ -226,16 +226,13 @@ SSLEngineResult unwrap(SslHandler handler, ByteBuf in, int readerIndex, int len,
             }
 
             @Override
-            int calculateOutNetBufSize(SslHandler handler, int pendingBytes) {
+            int calculateOutNetBufSize(SslHandler handler, int pendingBytes, int numComponents) {
                 return handler.maxPacketBufferSize;
             }
         };
 
         static SslEngineType forEngine(SSLEngine engine) {
-            if (engine instanceof OpenSslEngine) {
-                return TCNATIVE;
-            }
-            return JDK;
+            return engine instanceof ReferenceCountedOpenSslEngine ? TCNATIVE : JDK;
```

Comment:
eugh, I had it in my head that ReferenceCountedOpenSslEngine extends OpenSslEngine.